mypy/stubgenc.py

```diff
@@ -508,12 +508,14 @@ def is_classmethod(self, class_info: ClassInfo, name: str, obj: object) -> bool:
             return inspect.ismethod(obj)
 
     def is_staticmethod(self, class_info: ClassInfo | None, name: str, obj: object) -> bool:
-        if self.is_c_module:
+        if class_info is None:
             return False
+        elif self.is_c_module:
+            raw_lookup = getattr(class_info.cls, "__dict__")  # noqa: B009
+            raw_value = raw_lookup.get(name, obj)
+            return type(raw_value).__name__ in ("staticmethod")
```

Comment:
Should this be an equality check?

Reply:
yes, we discussed this many months ago: https://github.com/python/mypy/pull/14934#discussion_r1148561533

Reply:
Yes it should 🙈 I fixed it.